library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopEdge.java

```diff
@@ -127,29 +129,60 @@ public Vertex inVertex() {
         return inVertex;
     }
 
-    private static GafferPopVertex checkVertex(final Object vertexId, final GafferPopGraph graph) {
-        final GafferPopVertex gafferPopVertex;
-        if (vertexId instanceof Vertex) {
-            if (vertexId instanceof GafferPopVertex) {
-                gafferPopVertex = checkVertex(((GafferPopVertex) vertexId), graph);
-            } else {
-                gafferPopVertex = new GafferPopVertex(GafferPopGraph.ID_LABEL, ((Vertex) vertexId).id(), graph);
-            }
+    /**
+     * Gets the vertex ID object from the supplied vertex.
+     *
+     * @param vertex The vertex Object or ID.
+     * @return The ID for the vertex.
```

Comment:
This doesn't seem to reflect what the method does. It doesn't always return the id.

Reply:
I see. Could you add a little to the Javadoc to make this clearer.